File: Python/6py 2/sort_array.py
"""
|--------------------------------------------------------------------------
| Sort the odd
|--------------------------------------------------------------------------
|
"""

def sort_array(array):
   odd = sorted([x for x in array if x % 2 != 0])
   return [odd.pop(0) if x % 2 != 0 else x for x in array]


sort_array=lambda a:(lambda x:[next(x)if e%2 else e for e in a])(filter(2..__rmod__,sorted(a)))


def sort_array(source_array):
   result = sorted([l for l in source_array if l % 2 == 1])
   for index, item in enumerate(source_array):
      if item % 2 == 0:
         result.insert(index, item)
   return result


print(sort_array([5, 3, 2, 8, 1, 4]), [1, 3, 2, 8, 5, 4])
print(sort_array([5, 3, 1, 8, 0]), [1, 3, 5, 8, 0])
print(sort_array([]),[])

File: JavaScript/6js 3/sortArray.js
/*
|--------------------------------------------------------------------------
| Sort the odd
|--------------------------------------------------------------------------
|
*/

function sortArray(array) {
   const odd = array.filter((num) => num % 2 !== 0).sort((a, b) => a - b);
   return array.map((num) => (num % 2 !== 0 ? odd.shift() : num));
}


function sortArray(array) {
   for (let i = 0; i < array.length; i++) {
      for (let j = i + 1; j < array.length; j++) {
         if (array[i] % 2 != 0 && array[j] % 2 != 0 && array[j] < array[i]) {
            let tmp = array[i];
            array[i] = array[j];
            array[j] = tmp;
         }
      }
   }
   return array;
}


function sortArray(array) {
   const odds = array.filter((n) => n & 1).sort((a, b) => a - b);
   return array.map((n) => (n & 1 ? odds.shift() : n));
}


console.log(sortArray([5, 3, 2, 8, 1, 4]), [1, 3, 2, 8, 5, 4]);
console.log(sortArray([5, 3, 1, 8, 0]), [1, 3, 5, 8, 0]);
console.log(sortArray([]), []);